contrib/core-contract-tests/tests/pox-4/pox_DelegateStackIncreaseCommand.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  logCommand,
+  PoxCommand,
+  Real,
+  Stub,
+  Wallet,
+} from "./pox_CommandModel.ts";
+import { poxAddressToTuple } from "@stacks/stacking";
+import { expect } from "vitest";
+import { Cl } from "@stacks/transactions";
+
+/**
+ * The DelegateStackIncreaseCommand allows a pool operator to
+ * increase an active stacking lock, issuing a "partial commitment"
+ * for the increased cycles.
+ * 
+ * This method increases stacker's current lockup and partially
+ * commits the additional STX to pox-addr
+ *
+ * Constraints for running this command include:
+ * - The Stacker must have locked uSTX.
+ * - The Operator has to currently be delegated by the Stacker.
+ * - The increase amount must be greater than 0.
+ * - Stacker's unlocked uSTX amount must be greater than or equal
+ *   to the value of the increase amount.
+ * - Stacker's maximum delegated amount must be greater than or equal
+ *   to the final locked amount.
+ * - The Operator must have locked the Stacker's previously locked funds.
```

Comment:
The locking period must not end with the next cycle

Reply:
@friedger can you point me to the part of the PoX-4 contract that throws an error for this? Is it this [assertion](https://github.com/stacks-network/stacks-core/blob/feat/pox-4-stateful-delegate-stack-increase/stackslib/src/chainstate/stacks/boot/pox-4.clar#L1285-L1289)?

Reply:
@BowTiedRadone I meant this one: https://github.com/stacks-network/stacks-core/blob/feat/pox-4-stateful-delegate-stack-increase/stackslib/src/chainstate/stacks/boot/pox-4.clar#L1239

Reply:
Thank you!

Reply:
@friedger, should be now resolved in https://github.com/stacks-network/stacks-core/pull/4607/commits/b91848df01d1d91cb192eb3e424fba31105395b7